packages/next/export/worker.js

```diff
@@ -37,12 +37,24 @@ export default async function({
   }
 
   try {
-    let { query = {} } = pathMap
+    const { query: originalQuery = {} } = pathMap
     const { page } = pathMap
     const filePath = path === '/' ? '/index' : path
     const ampPath = `${filePath}.amp`
+    let query = { ...originalQuery }
     let params
 
+    // We need to show a warning if they try to provide query values
+    // for an auto-exported page since they won't be available
+    const hasOrigQueryValues = Object.keys(originalQuery).length > 0
+    const queryWithAutoExportWarn = () => {
+      if (hasOrigQueryValues) {
+        throw new Error(
+          `\nWarn: you provided query values for ${path} which is an auto-exported page. These can not be applied since the page can no longer be re-rendered on the server. To disable auto-export for this page add \`getInitialProps\`\n`
```

Comment:
Should say `Error` now not `Warn` — plus, this fails the command right (exit code 1)?

Reply:
Updated and also added check ensuring exit code 1